api/desecapi/serializers.py

```diff
@@ -22,14 +23,23 @@
 
 class CaptchaSerializer(serializers.ModelSerializer):
     challenge = serializers.SerializerMethodField()
+    kind = serializers.ChoiceField(
+        choices=['image', 'audio'],  # TODO this list could be automatically generated
+        default='image',
+    )
 
     class Meta:
         model = models.Captcha
-        fields = ('id', 'challenge') if not settings.DEBUG else ('id', 'challenge', 'content')
+        fields = ('id', 'challenge', 'kind') if not settings.DEBUG else ('id', 'challenge', 'kind', 'content')
 
     def get_challenge(self, obj: models.Captcha):
         # TODO Does this need to be stored in the object instance, in case this method gets called twice?
-        challenge = ImageCaptcha().generate(obj.content).getvalue()
+        if obj.kind == models.Captcha.Kind.IMAGE:
+            challenge = ImageCaptcha().generate(obj.content).getvalue()
+        elif obj.kind == models.Captcha.Kind.AUDIO:
+            challenge = AudioCaptcha().generate(obj.content)
```

Comment:
not sure why audio doesn't need `getvalue()`, but I trust you did it right

Reply:
yup